esp/src/src-react/components/controls/Grid.tsx

```diff
@@ -275,20 +285,40 @@ const FluentStoreGrid: React.FunctionComponent<FluentStoreGridProps> = ({
         columnWidths.set(column.key, newWidth);
     }, [columnWidths]);
 
-    return <DetailsList
-        compact={true}
-        items={items}
-        columns={fluentColumns}
-        setKey="set"
-        layoutMode={DetailsListLayoutMode.justified}
-        selection={selectionHandler}
-        isSelectedOnFocus={false}
-        selectionPreservedOnEmptyClick={true}
-        onColumnHeaderClick={onColumnClick}
-        onRenderDetailsHeader={renderDetailsHeader}
-        onColumnResize={columnResize}
-        styles={gridStyles(height)}
-    />;
+    //  Monitor scroll events  ---
+    const id = useId("fluent-store-grid-");
+    const detailListConponent = React.useRef<IDetailsListEx | null>(null);
+    const [detailListElement, setDetailListElement] = React.useState<HTMLDivElement | null>(null);
+    useMount(() => {
+        const detailListElement = document.querySelector<HTMLDivElement>(`#${id} .ms-DetailsList`);
+        setDetailListElement(detailListElement);
+        if (detailListConponent.current?._list?.current) {
+            detailListConponent.current._list.current._scrollElement = detailListElement;
+        }
+    });
+    useOnEvent(detailListElement, "scroll", () => {
```

Comment:
This fix works without the `_onScroll`. I noticed it seemed pretty slow, so I tested it with only the `_onAsyncScrollDebounced`, and it does properly update the list content.